lib/presentation/widgets/m3_widgets/custom_alert_dialog.dart

```diff
@@ -0,0 +1,48 @@
+import 'package:flutter/material.dart';
+import 'package:trip_tally/presentation/theme/app_colors.dart';
+import 'package:trip_tally/presentation/utils/enums/context_extensions.dart';
+
+class CustomAlertDialog extends StatelessWidget {
+  const CustomAlertDialog({
+    required this.title,
+    required this.info,
+    required this.isConfirmation,
+    required this.actionButtonText,
+    super.key,
+  });
+
+  final String title;
+  final String info;
+  final String actionButtonText;
+  final bool isConfirmation;
```

Comment:
The term isConfirmation can be misleading because the action button will always serve as a confirmation of some kind (whether it's confirming a deletion, saving data, etc.). Instead of that let's try to use: `isWarningAction`